### PR TITLE
feat(TPG >= 4.64)!: added cmek support

### LINKED ISSUE #9: IP Address?
Any possible way to get the ip_address of the primary instance? 🤔 
I know the outputs are not part of the module but just wanted to see if that was possible

### LINKED ISSUE #11: Read CPU Count
I was trying to pass in a variable to control CPU count on read instance but doesn't work. It appears the resource doesn't take in a  _machine_cpu_count_ like the _primary_ resource

```
resource "google_alloydb_instance" "read_pool" {
  for_each      = local.read_pool_instance
  cluster       = google_alloydb_cluster.default.name
  instance_id   = each.key
  instance_type = each.value.instance_type

  read_pool_config {
    node_count = each.value.node_count
  }

  database_flags = each.value.database_flags

  depends_on = [google_alloydb_instance.primary]
}
```

### LINKED ISSUE #12: Add CMEK option in this module
Add [CMEK](https://github.com/hashicorp/terraform-provider-google/blob/main/CHANGELOG.md#4640-may-8-2023) option in this module for cluster and automayer backup.

### PR DESCRIPTION
**Pull request by imrannayer**
Fix #12 
Fix #11 
Fix #9

### CODE REVIEW

File: build/int.cloudbuild.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-timeout: 3600s
+timeout: 7200s
```
Comment (bharathkkb) [inline build/int.cloudbuild.yaml:15]: We can split into multiple build stages if tests can be run in parallel ([example](https://github.com/terraform-google-modules/terraform-google-sql-db/blob/master/build/int.cloudbuild.yaml))

**Review by bharathkkb — APPROVED** [review]
CI looks like it maybe a unrelated error `Error: Error waiting for Deleting Instance: Error code 13, message: an internal error has occurred`. You can also try adding retry config ([example](https://github.com/GoogleCloudPlatform/terraform-google-three-tier-web-app/blob/947f15b85c12b2f13283162ad460dfb33433d7fd/test/integration/simple_example/simple_example_test.go#L28-L33))

File: build/int.cloudbuild.yaml (hunk 1 above)
Comment (imrannayer) [reply build/int.cloudbuild.yaml:15]: I was just checking if we have a timeout issue. But I think in this case it is a problem with AloyDB APIs. They are throwing `internal error` on deleting cluster majority of the run. I will change timeout back to `3600`. Create and verify stages are always successful.

**Comment by imrannayer** [comment]
> CI looks like it maybe a unrelated error `Error: Error waiting for Deleting Instance: Error code 13, message: an internal error has occurred`. You can also try adding retry config ([example](https://github.com/GoogleCloudPlatform/terraform-google-three-tier-web-app/blob/947f15b85c12b2f13283162ad460dfb33433d7fd/test/integration/simple_example/simple_example_test.go#L28-L33))

I tried it but instance is going in a state where it is not deleting either through terraform or manually. Only way to fix it is shutdown the project.

**Pull request merged**